### PR TITLE
runtime(doc): Fill in a few details regarding :enums

### PR DESCRIPTION
**Pull request by zzzyxwvut**
- Mention the support of `eval()` for enumeration values.
- Clarify the extent of immutability for enumeration values.
- Specify the requirements for class methods to meet for  
  class variable initialisation and their use in nested  
  functions and lambda expressions.
- Remove a duplicate sentence that describes how to access  
  parent class methods in derivative classes (see another  
"copy" two paragraphs below).

### CODE REVIEW

File: runtime/doc/vim9class.txt [hunk 1]
```diff
@@ -972,8 +972,8 @@ The following example shows an enum with object variables and methods: >
     echo Planet.Earth.has_rings
 <
 						*E1421* *E1423* *E1424* *E1425*
-Enums and their values are immutable. They cannot be modified after
-declaration and cannot be utilized as numerical or string types.
+Enums and their values are immutable; however, mutable components are allowed
+for enum values.  They cannot be utilized as numerical or string types.
```
Comment (chrisbra) [inline runtime/doc/vim9class.txt]: I don't understand that sentence. First we are saying *... their values are immutable*  and just a few words further we are saying that *mutable components are allowed for enum values*? What does that mean and what exactly mutable for enum values?

Comment (zzzyxwvut) [reply runtime/doc/vim9class.txt]: Sigh... I wish I could express it better.

Well, given these types:

```
vim9script

class Class
    var text: string = "Class"
endclass

enum Enum 
    INSTANCE

    var value: number = 0

    def new()
    enddef

    def SetValue(value: number)
	this.value = value
    enddef
endenum
```

> Enums and their values are immutable

Neither enumerations are extendable by a class (and so no  
new state can be acquired through class inheritance),

```
# E1416
enum Enum extends Class
    INSTANCE
endenum
```

nor any _new_ instances beyond those already declared can be  
created.

```
# E1325
Enum.new()
```

But the state of any enumeration value can be changed when  
the said state is mutable.

```
Enum.INSTANCE.SetValue(5)
echo Enum.INSTANCE
```

Comment (zzzyxwvut) [reply runtime/doc/vim9class.txt]: I've rewritten vague parts.  Hopefully, the changed text is
more clear now.

**Review by yegappan — APPROVED** [review]

**Comment by chrisbra** [comment]
Thanks, looks good

**Pull request merged**